jwst/extract_1d/extract.py

```diff
@@ -1439,11 +1439,24 @@ def locn_from_wcs(
         fwd_transform = self.wcs(x, y)
         middle_wl = np.nanmean(fwd_transform[2])
 
-        try:
-            x_y = self.wcs.backward_transform(targ_ra, targ_dec, middle_wl)
-        except NotImplementedError:
-            log.warning("Inverse wcs is not implemented, so can't use target coordinates to get location of spectrum.")
-            return
+        if input_model.meta.exposure.type in ['NRS_FIXEDSLIT', 'NRS_MSASPEC',
+                                              'NRS_BRIGHTOBJ']:
+            if slit is None:
+                xpos = input_model.source_xpos
+                ypos = input_model.source_ypos
+            else:
+                xpos = slit.source_xpos
+                ypos = slit.source_ypos
+
+            slit2det = self.wcs.get_transform('slit_frame', 'detector')
+            x_y = slit2det(xpos, ypos, middle_wl)
```

Comment:
I wonder if we should have a log message in this new branch saying that the source x/y pos from the meta data is being used for centering, instead of the target RA/Dec. A user could run this with `use_source_posn` turned on, assuming it would compute the position from the RA/Dec and then not get that result.